File: energy-mgmt-login/src/main/java/ro/tucn/energy_mgmt_login/service/user/UserService.java
package ro.tucn.energy_mgmt_login.service.user;

import ro.tucn.energy_mgmt_login.dto.user.UserRequestDTO;
import ro.tucn.energy_mgmt_login.dto.user.UserResponseDTO;

import java.util.List;
import java.util.UUID;

public interface UserService {
    UserResponseDTO getInfo(UserRequestDTO userRequestDTO);
    List<UserResponseDTO> getAll();
    UserResponseDTO getByUsername(String username);
    UserResponseDTO save(UserRequestDTO userRequestDTO);
    UserResponseDTO deleteById(UUID id);
    UserResponseDTO updateById(UUID userId, UserRequestDTO userRequestDTO);
}

File: energy-mgmt-frontend/src/app/core/service/user/user.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import { UserModel } from "../../../shared/models/user.model";
import {environment} from "../../../../environments/environment.development"
@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient) {
  }

  getInfo(): Observable<UserModel> {
    return this.http.get<UserModel>('user/v1/info');
  }

  getAll(): Observable<UserModel[]> {
    return this.http.get<UserModel[]>('user/v1/all');
  }

  getByUsername(username: string): Observable<UserModel> {
    return this.http.get<UserModel>(`user/v1/${username}`);
  }

  insert(user: UserModel): Observable<UserModel> {
    return this.http.post<UserModel>(`user/v1/one`, user);
  }

  deleteById(userId: string): Observable<UserModel> {
    return this.http.delete<UserModel>(`user/v1/${userId}`);
  }

  update(user: UserModel, userId: string | undefined): Observable<UserModel> {
    return this.http.put<UserModel>(`user/v1/${userId}`, user);
  }

  // register(newUser: UserModel): Observable<UserModel> {
  //   return this.http.post<UserModel>(`user/v1/one`, newUser);
  // }
}
